Zotero/Scenes/General/NoteEditorCoordinator.swift

```diff
@@ -75,7 +75,15 @@ final class NoteEditorCoordinator: NSObject, Coordinator {
     func start(animated: Bool) {
         guard let dbStorage = controllers.userControllers?.dbStorage, let openItemsController = controllers.userControllers?.openItemsController else { return }
 
-        let state = NoteEditorState(kind: kind, library: library, title: title, text: initialText, tags: initialTags, openItemsCount: openItemsController.getItems(for: sessionIdentifier).count)
+        let state = NoteEditorState(
+            kind: kind,
+            library: library,
+            title: title,
+            text: initialText,
+            tags: initialTags,
+            openItemsCount: openItemsController.getItems(for: sessionIdentifier).count,
+            activityTitle: kind.key.flatMap { (try? dbStorage.perform(request: ReadItemDbRequest(libraryId: library.identifier, key: $0), on: .main))?.displayTitle }
```

Comment:
You don't need to re-read this from database. When showing a note you already start with an RItem (or Note), so you can just pass activityTitle from that when creating NoteEditorCoordinator. So you'd add activityTitle parameter to DetailCoordinator.showNote and I see we call showNote in ItemDetailViewController and ItemsViewController, where you have either RItem or Note available.

Reply:
There are cases now where the Note is not available, just individual parameters, but let me check if I can refactor this up the chain.

Reply:
Check https://github.com/zotero/zotero-ios/pull/829/commits/008228c3f8c93d7d76003a5a6286f34cf6961d03
I selected to rename `title: NoteEditorState.TitleData?` to `parentTitleData` and `activityTitle: String?` to simply `title` for further clarification.

Reply:
Looks good now, I guess you can rename the `activityTitle` in PDF reader too, not sure whether it conflicted with another title there as well.